### ISSUE TITLE
footnote text missing

### ISSUE THREAD

**Issue by jniggemann** [tomchristie/mkdocs#329]
This is related to #294: When using the footnotes extension, the actual footnote texts are missing. 
Example markdown source:
```
This is normal text with a footnote[^2]
Another line of normal text

[^2]: This is the footnote text
```
This is rendered as follows:
```
This is normal text with a footnote²
Another line of normal text
```
The superscript number to indicate a footnote is properly rendered, but the actual text and the link target `This is the footnote text` is nowhere to be found.

**Comment by d0ugal** [comment]
To expand on #294's relevancy slightly. It has a partial fix for the issue (I've not actually verified this) but it breaks it for anyone not using the footnotes extension (or it at least break the tests which don't have it enabled). However, it should make for a good start for anyone looking to write a fix.

**Comment by jniggemann** [comment]
The main issue seems to stem from the fact that the footnotes extension always appends its HTML content the the very end of the generated HTML (that's what @jdhoek meant I suppose).

All we need to do now is change line 28 of toc.py to not *append* the TOC tag and delimiter at the end, but *prepend* them to the markdown source:
```python
    return '\n[TOC]' + '\n\n' + TOC_DELIMITER + markdown_content
```
This effectively reverses html_content and toc_html coming from `post_process`, so we also need to reverse them in line 45 of build.py:
```python
    (toc_html, html_content) = toc.post_process(html_content)
```

PR follows.

**Comment by jniggemann** [comment]
Oups, did't want to close the issue...

**Comment by jniggemann** [comment]
Fixed by PR #331

**Comment by d0ugal** [comment]
Closing this as #331 is merged.
